schema/snapcraft.yaml

```diff
@@ -66,6 +66,18 @@ definitions:
 title: snapcraft schema
 type: object
 properties:
+  vendoring:
+    type: array
+    validation-failure:
+      "a list of valid host names needs to be provided."
+    description: external sources available to resolve dependencies
+    minItems: 1
+    uniqueItems: true
+    items:
+      - type: string
+        validation-failure:
+          "{.instance!r} is not a valid hostname."
+        format: hostname
```

Comment:
This prevents the use of any sort of wildcard domain, e.g. `.ubuntu.com` to cover both `archive.ubuntu.com` and `security.ubuntu.com`. Squid allows such things, should we?